tflint/tflint.go

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package tflint
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/terraform-linters/tflint/cmd"
+)
+
+func RunTflintWithOpts(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
+	configFile, err := findTflintConfigInProject(terragruntOptions)
+	if err != nil {
+		return err
+	}
+	terragruntOptions.Logger.Debugf("Found .tflint.hcl file in %s", configFile)
+
+	variables, err := inputsToTflintVar(terragruntConfig.Inputs)
+	if err != nil {
+		return err
+	}
+
+	terragruntOptions.Logger.Debugf("Initializing tflint in directory %s", terragruntOptions.WorkingDir)
+	cli := cmd.NewCLI(terragruntOptions.Writer, terragruntOptions.ErrWriter)
+
+	initArgs := []string{"tflint", "--init", "--config", configFile, terragruntOptions.WorkingDir}
+	statusCode := cli.Run(initArgs)
+	if statusCode != 0 {
+		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+	}
+
+	args := []string{"tflint"}
+	args = append(args, variables...)
+	args = append(args, "--config", configFile)
+	args = append(args, "--module")
+	args = append(args, terragruntOptions.WorkingDir)
+
+	terragruntOptions.Logger.Debugf("Running tflint with args %v", args)
+	statusCode = cli.Run(args)
+
+	if statusCode == cmd.ExitCodeError {
+		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+	} else if statusCode == cmd.ExitCodeIssuesFound {
+		// When tflint finds issues in the project, it prints out but does not stop execution of Terragrunt.
+		terragruntOptions.Logger.Warnf("Tflint found issues in the project. Check for the tflint logs.")
```

Comment:
If tflint returned an error, shouldn't we return an error?

Reply:
Fixed 6394523